src/client/Microsoft.Identity.Client/Cache/Items/MsalAccountCacheItem.cs

```diff
@@ -202,7 +211,8 @@ internal static MsalAccountCacheItem FromJObject(JObject j)
                 LocalAccountId = JsonHelper.ExtractExistingOrEmptyString(j, StorageJsonKeys.LocalAccountId),
                 AuthorityType = JsonHelper.ExtractExistingOrEmptyString(j, StorageJsonKeys.AuthorityType),
                 TenantId = JsonHelper.ExtractExistingOrEmptyString(j, StorageJsonKeys.Realm),
-                WamAccountIds = JsonHelper.ExtractInnerJsonAsDictionary(j, StorageJsonKeys.WamAccountIds)
+                WamAccountIds = JsonHelper.ExtractInnerJsonAsDictionary(j, StorageJsonKeys.WamAccountIds),
+                AccountSource = JsonHelper.ExtractExistingOrEmptyString(j, StorageJsonKeys.AccountSource)
```

Comment:
Why is this added to the cached item? Is this needed when retrieving from cache?

Reply:
Yes this is need to retreive the property to decide between broker and non broker flow in silent request.